Caching implmented using reedis

obedjamir opened the pull request:
getStatusByName api implemented with @Cacheable
Cache time to live set to 5 seconds
test cases iplemented for cacheable

obedjamir committed:
Caching implmented using reedis
getStatusByName api implemented with @Cacheable
Cache time to live set to 5 seconds
test cases iplemented for cacheable

PinakiGhosh commented on core/src/main/java/com/bankbazaar/kafka/core/model/FileStatusEntity.java:
```diff
@@ -16,7 +16,7 @@ public class FileStatusEntity {
     private Long id;
 
-    @Column(name="file_name",nullable = false)
+    @Column(name="fileName",nullable = false)
```
Why is this change required?

obedjamir replied:
I have reverted the change

PinakiGhosh commented on core/src/main/java/com/bankbazaar/kafka/core/model/Status.java:
```diff
@@ -1,5 +1,5 @@
 
 public enum Status {
-    NEW,IN_PROGRESS,SUCCESS,FAILURE,ERROR
+    NEW,IN_PROGRESS,SUCCESS,FAILURE,ERROR,NULL
```
NULL should not be a status

obedjamir replied:
NULL removed

PinakiGhosh commented on core/src/main/java/com/bankbazaar/kafka/core/manager/FileStatusManager.java:
```diff
@@ -34,4 +35,13 @@ public Optional<FileStatusEntity> getEntry(Long id)
+        FileStatusEntity response = fileStatusRepository.findByFileName(name);
+        if(response==null)
+        {
+            return Status.NULL;
```
Just return null

obedjamir replied:
method now returns null

PinakiGhosh commented on core/src/main/java/com/bankbazaar/kafka/core/manager/FileStatusManager.java:
```diff
@@ -34,4 +35,13 @@ public Optional<FileStatusEntity> getEntry(Long id)
     }
+    public Status getEntry(String name)
+    {
+        FileStatusEntity response = fileStatusRepository.findByFileName(name);
```
Make the repository return an Optional

obedjamir replied:
repository returns optional

PinakiGhosh commented on service/src/main/java/com/bankbazaar/kafka/service/config/CacheConfig.java:
```diff
@@ -0,0 +1,20 @@
+    public RedisCacheManagerBuilderCustomizer redisCacheManagerBuilderCustomizer() {
+        return (builder) -> builder
+                .withCacheConfiguration("StatusCache",
+                        RedisCacheConfiguration.defaultCacheConfig().entryTtl(Duration.ofSeconds(5)));
```
Add the ttl as a config. This way you can have separate values for actual application and test

obedjamir replied:
Ttl added as config

PinakiGhosh commented on service/src/main/java/com/bankbazaar/kafka/service/service/RedisUtil.java:
```diff
@@ -14,7 +14,7 @@ public class RedisUtil {
     public void saveToRedis(Long id, Status status)
     {
-        redisTemplate.opsForValue().set(id, status, 10, TimeUnit.SECONDS);
+        redisTemplate.opsForValue().set(id, status, 5, TimeUnit.SECONDS);
```
Why is this required?

obedjamir replied:
This was done to minimize time during testing. However, ttl can now be configured seperately

PinakiGhosh commented on service/src/test/java/com/bankbazaar/kafka/service/controller/KafkaControllerTest.java:
```diff
@@ -90,10 +103,16 @@ void userController() throws Exception {
         consumeBadApi(dataDto4);
 
-        Thread.sleep(10000);
+        Thread.sleep(5000);
```
You should not be using sleep anywhere

obedjamir replied:
this has been removed

obedjamir committed:
Requested Changes Implemented
Status.NULL removed
Returns null instead
Repository returns optional
Ttl added as config

obedjamir committed:
TTL config added for test

PinakiGhosh commented on service/src/test/java/com/bankbazaar/kafka/service/controller/KafkaControllerTest.java:
```diff
@@ -46,7 +45,10 @@
+        /**
+         * controller returns null before any db entry
+         */
+        assertNull(consumeApi("test2.csv"));
```
This is a scenario where the given filename does not exist right? Should it not then respond with a 404 status?

PinakiGhosh commented on service/src/main/resources/application.properties:
```diff
@@ -30,4 +30,4 @@ spring.cloud.stream.bindings.consumer-out-0.destination=Notification
 spring.datasource.maxRetries=1
 spring.cache.type=redis
-
+spring.datasource.timeToLive=259200
```
Mention the unit as well in the properties

fileStatus.cache.expiry.inseconds=

Avoid using spring.* since those have are generally for the spring boot framework

obedjamir committed:
Test Case Implementation for Redis Cache
@Cacheable and @CacheEvict implemented in Manager layer
Test case implemented for each status in redis cache

obedjamir committed:
Controller and respective tests updated
Return code 404 if entry not found
